core/src/test/java/org/apache/accumulo/core/client/rfile/RFileClientTest.java

```diff
@@ -840,4 +842,35 @@ public void testMultipleFilesAndCache() throws Exception {
     assertEquals(testData, toMap(scanner));
     scanner.close();
   }
+
+  @Test
+  public void testFileSystemFromUri() throws Exception {
+    String localFsClass = "LocalFileSystem";
+
+    String fileUri = "hdfs://127.0.0.5:8080/bulk-xyx/file1.rf";
+    // There was a bug in the code where the default hadoop file system was always used. This test
+    // checks that the hadoop filesystem used it based on the URI and not the default filesystem. In
+    // this env the default file system is the local hadoop file system.
+    var exception = assertThrows(IOException.class, () -> RFile.newWriter().to(fileUri).build());
+    // Ensure the DistributedFileSystem was used.
+    assertTrue(Arrays.stream(exception.getStackTrace())
+        .anyMatch(ste -> ste.getClassName().contains(DistributedFileSystem.class.getName())));
+    assertTrue(Arrays.stream(exception.getStackTrace())
+        .noneMatch(ste -> ste.getClassName().contains(localFsClass)));
+
+    var exception2 = assertThrows(RuntimeException.class, () -> {
+      var scanner = RFile.newScanner().from(fileUri).build();
+      scanner.iterator();
+    });
```

Comment:
Same specific check here:
```suggestion
    });
    assertTrue(exception2.getMessage().contains("to " + remoteFsHost
            + " failed on connection exception: java.net.ConnectException: Connection refused"));
```